pkg/middleware/recovery/recovery_test.go

```diff
@@ -0,0 +1,210 @@
+package recovery
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/test/bufconn"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+func TestPanic(t *testing.T) {
+	logger := logger.MustNewLogger("text", "info", "unix")
+	handlerFunc := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		panic("Unexpected error!")
+	})
+
+	t.Run("With panic Recovery", func(t *testing.T) {
+		handler := Panic(handlerFunc, logger)
+
+		req, err := http.NewRequest(http.MethodGet, "/", nil)
+		if err != nil {
+			require.NoError(t, err)
+		}
+
+		resp := httptest.NewRecorder()
+		handler.ServeHTTP(resp, req)
+
+		require.Equal(t, http.StatusInternalServerError, resp.Code)
+		require.Equal(t, http.StatusText(http.StatusInternalServerError), strings.Trim(resp.Body.String(), "\n"))
+	})
+}
+
+func TestUnaryPanicInterceptor(t *testing.T) {
+	listner := bufconn.Listen(1024 * 1024)
+
+	serverOpts := []grpc.ServerOption{
+		grpc.ChainUnaryInterceptor(
+			[]grpc.UnaryServerInterceptor{
+				UnaryPanicInterceptor(logger.MustNewLogger("text", "info", "unix")),
+			}...,
+		),
+	}
+
+	t.Run("With Unary Panic Interceptor", func(t *testing.T) {
+		srv := grpc.NewServer(serverOpts...)
+		t.Cleanup(func() {
+			srv.Stop()
+		})
+
+		openfgav1.RegisterOpenFGAServiceServer(srv, &unimplementedOpenFGAServiceServer{})
+
+		go func() {
+			err := srv.Serve(listner)
+			if err != nil {
+				t.Fail()
+			}
+		}()
+
+		dialer := func(context.Context, string) (net.Conn, error) {
+			return listner.Dial()
+		}
+
+		ctx, cancel := context.WithCancel(context.Background())
+		t.Cleanup(func() {
+			cancel()
+		})
+
+		opts := []grpc.DialOption{
+			grpc.WithContextDialer(dialer),
+			grpc.WithTransportCredentials(insecure.NewCredentials())}
+
+		conn, err := grpc.DialContext(ctx, "", opts...)
+		require.NoError(t, err)
+
+		t.Cleanup(func() {
+			conn.Close()
+		})
+
+		cli := openfgav1.NewOpenFGAServiceClient(conn)
+
+		_, err = cli.Read(ctx, &openfgav1.ReadRequest{})
+		require.ErrorContains(t, err, "Read not implemented")
+
+		_, err = cli.Check(ctx, &openfgav1.CheckRequest{})
+		require.ErrorContains(t, err, http.StatusText(http.StatusInternalServerError))
+
+		_, err = cli.ReadAuthorizationModels(ctx, &openfgav1.ReadAuthorizationModelsRequest{})
+		require.ErrorContains(t, err, "method ReadAuthorizationModels not implemented")
+	})
+}
+
+func TestStreamPanicInterceptor(t *testing.T) {
+	listner := bufconn.Listen(1024 * 1024)
+	t.Cleanup(func() {
```

Comment:
```suggestion
	t.Cleanup(listner.Close)
```

Reply:
cannot use listner.Close() (value of type error) as func() value in argument to t.Cleanup